storage/transfermanager/downloader.go

```diff
@@ -402,6 +499,40 @@ func (in *DownloadObjectInput) downloadShard(client *storage.Client, timeout tim
 	return
 }
 
+// DownloadDirectoryInput is the input for a directory to download.
+type DownloadDirectoryInput struct {
+	// Required fields.
+	Bucket         string
+	LocalDirectory string
```

Comment:
This needs a doc

Reply:
Looks good but I think we should add here the note about not modifying the directory while the download is in progress.